clippy_lints/src/matches.rs

```diff
@@ -914,19 +914,36 @@ fn check_match_single_binding(cx: &LateContext<'_, '_>, ex: &Expr<'_>, arms: &[A
     let mut applicability = Applicability::MaybeIncorrect;
     match arms[0].pat.kind {
         PatKind::Binding(..) | PatKind::Tuple(_, _) | PatKind::Struct(..) => {
+            let sugg;
+            let target_span;
+            // If this match is in a local (`let`) stmt
+            if let Some(parent_let_node) = opt_parent_let(cx, ex) {
```

Comment:
```suggestion
            // If this match is in a local (`let`) stmt
            let (target_span, sugg) = if let Some(parent_let_node) = opt_parent_let(cx, ex) {
```

and then return this tuple in the `if`/`else`